### ISSUE TITLE
Update to use Macros 1.1

### ISSUE THREAD

**Issue by dtolnay** [aldanor/pod-typeinfo#2]
Macros 1.1 is going to be stabilizing in Rust 1.15 in February. It provides a more composable way to implement traits for data structures:

```rust
#[macro_use]
extern crate typeinfo_derive;

#[derive(Debug, TypeInfo)]
pub struct Color { r: u16, g: u16, b: u16 }

#[derive(Debug, TypeInfo)]
#[repr(packed)]
pub struct Palette {
    monochrome: bool,
    colors: [Color; 16],
}
```

Currently if I wanted to use `typeinfo` and another crate which has its own equivalent of the `def!` macro, there is no way for me to use both on the same struct.

[Here is the RFC](https://github.com/rust-lang/rfcs/blob/master/text/1681-macros-1.1.md) describing the Macros 1.1 feature and [here is the tracking issue](https://github.com/rust-lang/rust/issues/35900) for stabilization.

**Comment by aldanor** [comment]
Hi @dtolnay, thanks for the tip -- yes, this will make it a lot cleaner if it works out; in fact, custom derive seems like a perfect fit for this kind of thing.

Has all of the macros 1.1 stuff landed in nightly already?

**Comment by dtolnay** [comment]
Yes it's been in nightly since September. There isn't good documentation yet but here is an example of how it works: https://github.com/dtolnay/syn#complete-example. Another good resource is [my SF meetup talk](https://air.mozilla.org/rust-meetup-december-2016-12-15/) (start the video at 3:00).
